compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -214,47 +213,25 @@ impl<'tcx> Const<'tcx> {
         Self::from_bits(tcx, n as u128, ty::TypingEnv::fully_monomorphized(), tcx.types.usize)
     }
 
-    /// Panics if self.kind != ty::ConstKind::Value
-    pub fn to_valtree(self) -> (ty::ValTree<'tcx>, Ty<'tcx>) {
+    /// Panics if `self.kind != ty::ConstKind::Value`.
+    pub fn to_value(self) -> ty::Value<'tcx> {
         match self.kind() {
-            ty::ConstKind::Value(ty, valtree) => (valtree, ty),
+            ty::ConstKind::Value(cv) => cv,
             _ => bug!("expected ConstKind::Value, got {:?}", self.kind()),
         }
     }
 
-    /// Attempts to convert to a `ValTree`
-    pub fn try_to_valtree(self) -> Option<(ty::ValTree<'tcx>, Ty<'tcx>)> {
+    /// Attempts to convert to a value.
+    pub fn try_to_value(self) -> Option<ty::Value<'tcx>> {
         match self.kind() {
-            ty::ConstKind::Value(ty, valtree) => Some((valtree, ty)),
+            ty::ConstKind::Value(cv) => Some(cv),
             _ => None,
         }
     }
 
-    #[inline]
-    pub fn try_to_scalar(self) -> Option<(Scalar, Ty<'tcx>)> {
-        let (valtree, ty) = self.try_to_valtree()?;
-        Some((valtree.try_to_scalar()?, ty))
-    }
-
-    pub fn try_to_bool(self) -> Option<bool> {
-        self.try_to_valtree()?.0.try_to_scalar_int()?.try_to_bool().ok()
-    }
-
     #[inline]
     pub fn try_to_target_usize(self, tcx: TyCtxt<'tcx>) -> Option<u64> {
```

Comment:
Keeping this on `ty::Const` because there's a large amount of code that inspects array lengths that would be made worse by having to do `try_to_value().and_then(|val| val.try_to_target_usize(tcx))`?

Seems reasonable to me :thinking: can a comment be added to state that since it doesn't seem unlikely that someone might come across this and be confused why it's the way it is

Reply:
> Keeping this on `ty::Const` because there's a large amount of code that inspects array lengths that would be made worse by having to do `try_to_value().and_then(|val| val.try_to_target_usize(tcx))`?

Yeah, this is used all over the place, including codegen, mir opts and clippy lints.

> Seems reasonable to me 🤔 can a comment be added to state that since it doesn't seem unlikely that someone might come across this and be confused why it's the way it is

👍